circle-mlir/infra/overlay/prepare-venv

```diff
@@ -27,9 +27,30 @@ VENV_ACTIVATE=${DRIVER_PATH}/${VENV_NAME}/bin/activate
 # NOTE please use venv's python instead of python after `source activation`.
 VENV_PYTHON=${DRIVER_PATH}/${VENV_NAME}/bin/python
 
+# Minimum python requirment version
+PYTHON_REQUIRED_MAJOR=3
+PYTHON_REQUIRED_MINOR=10
+
+PYTHON_BIN=$(which python$PYTHON_REQUIRED_MAJOR.$PYTHON_REQUIRED_MINOR 2>/dev/null || true)
+if [ -z "$PYTHON_BIN" ]; then
+  PYTHON_BIN=$(which python3)
+fi
```

Comment:
(Optional)
You can replace `which` with `command -v` because it’s a POSIX‑standard builtin and more reliable than the external `which`.
```suggestion
if command -v python${PYTHON_REQUIRED_MAJOR}.${PYTHON_REQUIRED_MINOR} >/dev/null; then
  PYTHON_BIN=python${PYTHON_REQUIRED_MAJOR}.${PYTHON_REQUIRED_MINOR}
elif command -v python3 >/dev/null; then
  PYTHON_BIN=python3
else
  echo "Error: Python ${REQ_MAJOR}.${REQ_MINOR}+ not found"
  exit 1
fi
```